cpp/tests/test_math_time_series_functor.cpp

```diff
@@ -0,0 +1,80 @@
+#include "memilio/math/time_series_functor.h"
+#include "memilio/utils/random_number_generator.h"
+
+#include <gtest/gtest.h>
+
+class TestMathTimeSeriesFunctor : public ::testing::Test
+{
+public:
+    using TSF = mio::TimeSeriesFunctor<double>;
+
+    const int num_evals = 1000;
```

Comment:
I think the tests are more readable if the number of iterations is just hard coded there, doesn't matter if the values can be different in each test, different tests may require different iterations anyway.